### PR TITLE
Add milestone metadata

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by bingyanglin — DISMISSED** [review]

File: bee-message/src/error.rs [hunk 1]
```diff
@@ -193,6 +195,9 @@ impl fmt::Display for Error {
             Error::InvalidMetadataFeatureBlockLength(length) => {
                 write!(f, "invalid metadata feature block length {}", length)
```
Comment (Alex6323) [inline bee-message/src/error.rs]:
```suggestion
                write!(f, "invalid metadata feature block length {length}")
```

File: bee-message/src/error.rs [hunk 2]
```diff
@@ -193,6 +195,9 @@ impl fmt::Display for Error {
...
             }
+            Error::InvalidMilestoneMetadataLength(length) => {
+                write!(f, "invalid milestone metadata length {length}")
+            }
             Error::InvalidMigratedFundsEntryAmount(amount) => {
                 write!(f, "invalid migrated funds entry amount: {}", amount)
```
Comment (Alex6323) [inline bee-message/src/error.rs]:
```suggestion
                write!(f, "invalid migrated funds entry amount: {amount}")
```

File: bee-message/src/payload/milestone/essence.rs [hunk 3]
```diff
@@ -134,6 +152,9 @@ impl Packable for MilestoneEssence {
             verify_pow_scores(index, next_pow_score, next_pow_score_milestone_index).map_err(UnpackError::Packable)?;
         }
 
+        let metadata = BoxedSlicePrefix::<u8, MilestoneMetadataLength>::unpack::<_, VERIFY>(unpacker)
+            .map_packable_err(|err| Error::InvalidMilestoneMetadataLength(err.into_prefix_err().into()))?;
```
Comment (Alex6323) [inline bee-message/src/payload/milestone/essence.rs]:
```suggestion
            .map_packable_err(|e| Error::InvalidMilestoneMetadataLength(e.into_prefix_err().into()))?;
```
Sorry for being this nit-picky but down below you also use `e` or at least `_e`, so it's just for crate consistency I guess.

File: bee-message/src/payload/milestone/mod.rs [hunk 4]
```diff
@@ -217,6 +220,7 @@ pub mod dto {
                     .map_err(|_| DtoError::InvalidField("inclusionMerkleProof"))?;
                 let next_pow_score = value.next_pow_score;
                 let next_pow_score_milestone_index = value.next_pow_score_milestone_index;
+                let metadata = prefix_hex::decode(&value.metadata).map_err(|_e| DtoError::InvalidField("metadata"))?;
```
Comment (Alex6323) [inline bee-message/src/payload/milestone/mod.rs]:
```suggestion
                let metadata = prefix_hex::decode(&value.metadata).map_err(|_| DtoError::InvalidField("metadata"))?;
```
There's not much to criticise in this PR so you are forcing me to resort to this type of nit-picking :trollface: I think we normally just use `_` in such cases.

**Review by samuel-rufi — APPROVED** [review]

**Review by bingyanglin — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by pvdrz — APPROVED** [review]

**Pull request merged**